### PR TITLE
Add inference deactivation mode.

### LINKED ISSUE #86: Deactivate inference.
**Describe the task**
For experimenting purposes, it worths to measure the effectiveness of inference of the Annotator in reducing the required total number of `@NullUnmarked` to resolve all errors. To do that, a special mode is desired where it deactivates the inference procedure (no `@Nullable` injection). With this mode we can compare the total number of `@NullUnmarked` injections with inference activated \ deactivated.

**Expected behavior**
A flag to disable inference and resolve all errors with `@NullUnmarked` injections.

**Additional context**
This feature is __useful for experiments purposes only__.

### PR DESCRIPTION
**Pull request by nimakarimipour**
This PR resolves #86 by adding the specific analysis mode which deactivates inference of Annotator and all errors are resolved with `@NullUnmarked` annotations.

To enable this feature flag below must be passed:
```shell
-di, --deactivate-inference
```

Or 
```json
"INFERENCE_ACTIVATION": false
```

Please note: When inference is deactivated, force resolve mode is automatically activated to resolve all remaining errors with `@NullUnmarked` injection.
-----

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -50,7 +50,7 @@ tasks.test.dependsOn(':qual:publishToMavenLocal')
 
 // Set up environment variables for test configuration tu run with the latest development version.
 tasks.test.doFirst {
-    environment "NULLAWAY_TEST_VERSION", "0.10.0"
+    environment "NULLAWAY_TEST_VERSION", "0.10.2"
```
Comment (msridhar) [inline core/build.gradle:53]: How is this change related to the topic of this PR?

**Review by msridhar — COMMENTED** [review]
At a high level this PR makes sense, but can you comment on why you needed all the changes to update `config.log`?  Is that just for testing purposes, or is it important otherwise?

**Comment by nimakarimipour** [comment]
> At a high level this PR makes sense, but can you comment on why you needed all the changes to update `config.log`? Is that just for testing purposes, or is it important otherwise?

Yes that is required to make unit tests actually verify the injection of correct annotations. I wanted to make a separate PR for that but given that we had some PRs in the backlog I thought it might be better for this one specifically have it in one single PR. Please let me know if you prefer a separate PR for changes regarding `Log` class.

**Review by msridhar — APPROVED** [review]
👍
